src/components/Duration/DurationField.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import i18n from '@dhis2/d2-i18n'
+import { InputField } from '@dhis2/ui-core'
+import { Field } from '@dhis2/ui-forms'
+import { FormField } from '../'
+
+const durationRegex = /^(?!$)(\d+d)?(\d+h)?(\d+m)?(\d+s)?$/
```

Comment:
Thanks for adding the explanation!

The way this works right now requires the app to make this field required.
The only Component using the `DurationField` is the `LastUpdatedDuration` component.

Removing the `required` validator from `LastUpdatedDuration` breaks this in several ways:
1. Although optional `Invalid value` is displayed
2. Typing something then clearing the field produces a JS error

This would be a safe approach:
```js
const durationRegex = /^(\d+d)?(\d+h)?(\d+m)?(\d+s)?$/
const DURATION_VALIDATOR = duration =>
    !duration || duration.match(durationRegex)
        ? undefined
        : i18n.t('Invalid duration')
```

It works with empty values and the `required` validator takes care of empty values. Then there's no more need for the negative lookahead anymore.

Reply:
Nice catch. Fixed.